playbooks/cloud-provider-openstack-acceptance-test-flexvolume-cinder/run.yaml

```diff
@@ -0,0 +1,145 @@
+        PROJECTS: 'openstack/devstack-plugin-ceph'
+  tasks:
+    - shell:
+        cmd: |
+          set -x
+          # set -e
+          # set -o pipefail
+
+          # Create cloud-config
+          source /opt/stack/new/devstack/openrc admin admin
+          mkdir -p /etc/kubernetes/
+          cat << EOF >> /etc/kubernetes/cloud-config
+          [Global]
+          domain-id = $OS_PROJECT_DOMAIN_ID
+          tenant-name = $OS_TENANT_NAME
+          auth-url = $OS_AUTH_URL
+          password = $OS_PASSWORD
+          username = $OS_USERNAME
+          region = $OS_REGION_NAME
+
+          [BlockStorage]
+          bs-version = v2
+          ignore-volume-az = yes
+          EOF
+
+          export API_HOST_IP=$(ifconfig | awk '/^docker0/ {getline; print $2}' | awk -F ':' '{print $2}')
+          export KUBELET_HOST="0.0.0.0"
+          export ALLOW_SECURITY_CONTEXT=true
+          export ENABLE_CRI=false
+          export ENABLE_HOSTPATH_PROVISIONER=true
+          export ENABLE_SINGLE_CA_SIGNER=true
+          export KUBE_ENABLE_CLUSTER_DNS=false
+          export LOG_LEVEL=7
+          # DO NOT change the location of the cloud-config file. It is important for the old cinder provider to work
+          export CLOUD_CONFIG=/etc/kubernetes/cloud-config
+
+          # location of where the kubernetes processes log their output
+          mkdir -p '{{ ansible_user_dir }}/workspace/logs/kubernetes'
+          export LOG_DIR='{{ ansible_user_dir }}/workspace/logs/kubernetes'
+          # We need this for one of the conformance tests
+          export ALLOW_PRIVILEGED=true
+          # Just kick off all the processes and drop down to the command line
+          export ENABLE_DAEMON=true
+          export HOSTNAME_OVERRIDE=127.0.0.1
+          export MAX_TIME_FOR_URL_API_SERVER=5
+          export RUNTIME_CONFIG="settings.k8s.io/v1alpha1=true"
+
+          # Workaround to rebuild cinder-flex-volume-driver binary using patched code
+          # https://github.com/liu-sheng/cloud-provider-openstack/pull/17
+          cp -f pkg/flexvolume/cinder_client.go vendor/k8s.io/cloud-provider-openstack/pkg/flexvolume/cinder_client.go
+          # cp -f pkg/flexvolume/cinder_baremetal_util.go vendor/k8s.io/cloud-provider-openstack/pkg/flexvolume/cinder_baremetal_util.go
+          cd /home/zuul/src/k8s.io/cloud-provider-openstack && CGO_ENABLED=0 GOOS=linux go build \
+              -ldflags "-X 'main.version=19c9e833-dirty'" \
+              -o cinder-flex-volume-driver \
+              cmd/cinder-flex-volume-driver/main.go
+          cd -
+
+          mkdir -p /usr/libexec/kubernetes/kubelet-plugins/volume/exec/cinder.io~cinder-flex-volume-driver/
+          cp cinder-flex-volume-driver /usr/libexec/kubernetes/kubelet-plugins/volume/exec/cinder.io~cinder-flex-volume-driver/
+
+          export k8s_root="$GOPATH/src/k8s.io/kubernetes"
+          export kubectl="$k8s_root/cluster/kubectl.sh"
```

Comment:
If we do this, we should do this in all k8s jobs, so I prefer to make a refactor to all these k8s jobs in another PR since there are still so much repeated code worth to optimize.

Reply:
agree